### PR TITLE
docs(theme): Simplify README to follow best practices

### PR DESCRIPTION
**Pull request by lynnjepsen**

### CODE REVIEW

File: packages/mdc-theme/README.md [hunk 1]
```diff
@@ -46,13 +46,13 @@ npm install --save @material/theme
 
 ## Usage
 
-### Sass
+### Themeing
```
Comment (yeelan0319) [inline packages/mdc-theme/README.md]: Theming. IMHO, I think `Change them colors` feels less ambiguous

File: packages/mdc-theme/README.md [hunk 2]
```diff
@@ -46,13 +46,13 @@ npm install --save @material/theme
...
 
-#### Changing the theme colors
+MDC Theme mixins use CSS custom properties for browsers that support it, with a fallback to a pre-processed 
+static color if they don't. This enables runtime theming if CSS properties are available.
 
-MDC Theme makes it quite easy to change the theme colors for your application, and have the changes apply to all MDC-Web
-components. Simply define the three theme color variables before importing `mdc-theme` or any MDC-Web components that rely
-on it:
+To set the theme colors for your application, simply define the three theme color variables before importing
```
Comment (yeelan0319) [inline packages/mdc-theme/README.md]: Define or override? I feel the latter is more appropriate as we do provide default theme color. How about:

> MDC Theme makes it easy to develop your brand colors. You can simply override the default theme color by through Sass variable or CSS custom properties. In the latter case, it enables runtime theming.
> 
> **A note for Sass variables**, you need you define the three theme color variables before importing `mdc-theme` or any MDC-Web components that rely on it, like following:

File: packages/mdc-theme/README.md [hunk 3]
```diff
@@ -36,8 +38,6 @@ In addition, it's important to consider the style of text:
 to establish a visual identity and color many parts of your application. The latter refers to the style of text
 that is most prominent (low opacity, high contrast), and used to display most content.
 
-The text contrast colors are automatically calculated at the Sass level and exposed as part of this module.
```
Comment (yeelan0319) [inline packages/mdc-theme/README.md]: We have several issue regarding this, worth to provide a paragraph explaining this.
#608 #755

File: packages/mdc-theme/README.md [hunk 4]
```diff
@@ -64,205 +64,96 @@ $mdc-theme-background: #fff;
...
+| `--mdc-theme-text-icon-on-dark` | Icons on top of a dark-colored background. |
+
+### Sass Mixins, Variables, and Functions
+
+| Mixin                                           | Description |
+| ----------------------------------------------- | - |
+| `mdc-theme-prop($property, $style, $important)` | Applies a theme color to a property |
+| `mdc-theme-dark($root-selector, $compound)` | Creates a rule that is applied when the root element is within an Dark Theme context. If using the mixin on anything other than the base selector, you need to specify the base selector as a parameter. If using the mixin with a modifier class, pass `true` for the second argument. |
+
+#### Properties
```
Comment (yeelan0319) [inline packages/mdc-theme/README.md]: What are these properties and how can I use them? I don't think this clearly explained here.

**Comment by lynnjepsen** [comment]
Updated README based on comments, PTAL.

File: packages/mdc-theme/README.md [hunk 5]
```diff
@@ -16,28 +16,25 @@ path: /catalog/theme/
   </a>
 </div>-->
```
Comment (yeelan0319) [inline packages/mdc-theme/README.md]: How about:

> MDC Theme is a foundational module that theming MDC-Web components. 
> 
> The colors in this module are derived from three theme colors:
> - Primary: the primary color used in your application. This applies to a number of UI elements, such as app bars.
> - Accent: the accent color used in your application. This applies to UI elements such as FABs.
> - Background: the background color for your application. This is the color on top of which your UI is drawn.
> 
> and five text colors:
> - Primary, used for most text.
> - Secondary, used for text which is lower in the visual hierarchy.
> - Hint, used for text hints (such as those in text fields and labels).
> - Disabled, used for text in disabled components and content.
> - Icon, used for icons.
> 
> MDC-web themes the component using the combination of `mdc-theme--<theme>` for color and `mdc-theme-text-<text>-on-<theme>` for text on it. Users can apply the css modifiers to change the theme or the components.
>
> **Note: Dark theme**
> All components by defaults assume it is used on a light colored background (by default #FFF), but some of them are made aware of a Dark Theme context, aka placed on top of a dark background and provide a corresponding looking. There are two ways to make the components theme themselves in their dark mode: The first is to add `mdc-theme--dark` to a *container* element, which hold the component. The second way is to add `<component_name>--theme-dark` modifier class to the actual component element.

**Comment by lynnjepsen** [comment]
I went with

Intro paragraph. More about how we define theme color and what we provided.
How to change theme color, using css custom properties and sass
CSS modifiers available out of box
How to define your theming rules using sass mixins

I think almost all users will want to know how to change theme color, so it seemed more important than CSS modifiers.

And I rewrote the intro paragraph with your suggestions.

File: packages/mdc-theme/README.md [hunk 6]
```diff
@@ -16,27 +16,33 @@ path: /catalog/theme/
...
 
-MDC Theme is a foundational module that provides theming to MDC-Web components, and also makes it available to
-developers as Sass functions and mixins, as CSS custom properties, and as a set of CSS classes.
+This color palette comprises primary and accent colors that can be used for illustration or to develop your brand colors.
```
Comment (yeelan0319) [inline packages/mdc-theme/README.md]: super nit: feel if this sentence is repeating itself with the next one. Feel free to neglect if you disagree.

Comment (lynnjepsen) [inline packages/mdc-theme/README.md]: I took it straight from the Material Design guidelines page, so I'd like to leave it. I think it provides a link between the guidelines and are code.

And then I realized we didn't have a link to the guidelines! So I added that.

**Pull request merged**